# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/basetypes/Accessible.h
# Commit: 86da8f2a28ea
# Full Hash: 86da8f2a28ea1992671d08dfb56c881a1a82daa5
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-08-04 04:34:13
# Regressor Bug: 1781536
# File Overlap Count: 1
# Description:
#   Bug 1781536 part 3: Move RelationByType to base Accessible. r=morgan
#   
#   This required changing the RemoteAccessible implementations to return Relation instead of an array of RemoteAccessible.
#   Platform implementations have been updated to use the unified method where appropriate.
#   
# ==============================================================================

diff -r 95ae96802dbe -r 86da8f2a28ea accessible/basetypes/Accessible.h
--- a/accessible/basetypes/Accessible.h	Wed Aug 03 23:58:51 2022 +0000
+++ b/accessible/basetypes/Accessible.h	Wed Aug 03 23:58:51 2022 +0000
@@ -23,6 +23,8 @@
 class AccGroupInfo;
 class HyperTextAccessibleBase;
 class LocalAccessible;
+class Relation;
+enum class RelationType;
 class RemoteAccessible;
 class TableAccessibleBase;
 class TableCellAccessibleBase;
@@ -327,6 +329,11 @@
 
   virtual Maybe<int32_t> GetIntARIAAttr(nsAtom* aAttrName) const = 0;
 
+  /**
+   * Get the relation of the given type.
+   */
+  virtual Relation RelationByType(RelationType aType) const = 0;
+
   // Methods that interact with content.
 
   virtual void TakeFocus() const = 0;